src/Http/Controllers/ForumController.php

```diff
@@ -12,12 +14,21 @@ class ForumController extends Controller
      *
      * @return \Illuminate\View\View
      */
-    public function index()
+    public function index(Request $request)
     {
         $groups = Group::all();
 
-        $discussions = Discussion::orderBy('created_at', 'desc')
-            ->paginate(config('firefly.pagination.discussions'));
+        $discussionQuery = Discussion::orderBy('created_at', 'desc');
+        if (config('firefly.features.watchers') && $request->user()) {
+            $discussionQuery->leftJoin('discussion_user', function (JoinClause $join) use ($request) {
+                $join->on('discussion_user.discussion_id', '=', 'discussions.id')
+                    ->where('discussion_user.user_id', $request->user()->id);
+            })->select([
+                'discussions.*',
+                'discussion_user.user_id as is_being_watched',
+            ]);
+        }
+        $discussions = $discussionQuery->paginate(config('firefly.pagination.discussions'));
```

Comment:
Here is something that's a bit cleaner. I'm also curious if we could put the same `watchers` relation on the discussion? That would clean this up even more IMO.
```suggestion
        $discussions = Discussion::query()
            ->when(config('firefly.features.watchers') && $user = $request->user(), function ($query) use ($user) {
                $query->leftJoin('discussion_user', function (JoinClause $join) use ($user) {
                    $join->on('discussion_user.discussion_id', '=', 'discussions.id')
                        ->where('discussion_user.user_id', $user->id);
                })->select([
                    'discussions.*',
                    'discussion_user.user_id as is_being_watched',
                ]);
            })
            ->orderBy('created_at', 'desc')
            ->paginate(config('firefly.pagination.discussions'));
```

Reply:
I updated the logic to incorporate your changes, and then further cleaned up by adding as a query scope on the Discussion model. 

Was not 100% sure what you meant by adding to the watchers relation on the discussion. 
We have a:
```php
public function watchers(): BelongsToMany
{
    return $this->belongsToMany(User::class);
}
```